### PR TITLE
feat: add the ability to define list of valid scopes

### PR DESCRIPTION
**Pull request by churkin**
-----
[View rendered README.md](https://github.com/churkin/semantic-pull-requests/blob/master/README.md)

### CODE REVIEW

**Comment by zeke** [comment]
Thanks for the PR! I just merged #46 which introduced some conflicts here.  Would you mind resolving those?

**Comment by clakech** [comment]
WANT !!!

**Comment by churkin** [comment]
@zeke Sure!

File: __tests__/handle-pull-request-change.js [hunk 1]
```diff
@@ -32,6 +32,118 @@ describe('handlePullRequestChange', () => {
     expect(mock.isDone()).toBe(true)
   })
 
+  describe('when the message have a scope', () => {
```
Comment (zeke) [inline __tests__/handle-pull-request-change.js]:
```suggestion
  describe('custom scopes', () => {
```

File: __tests__/is-semantic-message.js [hunk 2]
```diff
@@ -13,3 +13,9 @@ test('returns false on bad input', () => {
   expect(isSemanticMessage(null)).toBe(false)
   expect(isSemanticMessage('non-semantic commit message')).toBe(false)
 })
+
+test.only('should check message scope', () => {
```
Comment (zeke) [inline __tests__/is-semantic-message.js]:
```suggestion
test('should check message scope', () => {
```

File: lib/is-semantic-message.js [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+
   if (error) {
     console.error(error)
     return false
   }
 
   const [result] = commits
-  return commitTypes.includes(result.header.type)
+  const { scope, type } = result.header
+  const isScopeValid = !scopes || !scope || scopes.indexOf(scope) !== -1
```
Comment (zeke) [inline lib/is-semantic-message.js]:
```suggestion
  const isScopeValid = !scopes || !scope || scopes.includes(scope)
```

**Comment by zeke** [comment]
This looks good, and I think it's almost ready to ship. Just one more request: I know you have the YML configuration documented in the README, but can you add something to the prose in the README about what scopes are, and how they can be used?

**Comment by clakech** [comment]
@zeke could you merge this one and I could add the prose in the README after ? or maybe @churkin can do it directly here ?

**Comment by zeke** [comment]
I went ahead and shipped this. It should auto-deploy to production soon.

If @clakech or @churkin wants to follow up with some README docs for this, that would be 💖

**Pull request merged**